src/tools/grpc_server.cc

```diff
@@ -243,6 +246,40 @@ static std::string pat(const std::shared_ptr<RE2> &p) {
     return p->pattern();
 }
 
+void CodeSearchImpl::TagsFirstSearch_(::CodeSearchResult* response, query& q, string& line_pat, match_stats& stats) {
+    string regex;
+    int32_t max_matches = q.max_matches;  // remember original value
+
+    add_match cb(response);
+
+    /* To surface the most important matches first, start with tags.
+       First pass: is the pattern an exact match for any tags? */
+    regex = "^" + line_pat + "$";
+    q.line_pat.reset(new RE2(regex, q.line_pat->options()));
+    run_tags_search(q, tagdata_, cb, tagmatch_, stats);
+
+    q.max_matches = max_matches - cb.match_count();
```

Comment:
`q.max_matches -= cb.match_count()` would be more idiomatic and I think clearer

Reply:
I'll improve the naming here! This isn't a decrement; it's a throwing out of the old value of `q.max_matches` because our match count is new. I'll switch `max_matches` to `original_max_matches` to hopefully make things clear.